source/collections/RecycleBin.ooc

```diff
@@ -0,0 +1,59 @@
+/*
+* Copyright (C) 2015 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with This software. If not, see <http://www.gnu.org/licenses/>.
+*/
+import threading/Mutex
+import VectorList
+
+RecycleBin: class <T> {
+	_mutex := Mutex new()
+	_free: Func (T)
+	//TODO: Replace with SynchronizedVectorList
+	_list := VectorList<T> new()
+	_maxCount: Int
+	init: func (=_maxCount, =_free)
+	free: override func {
+		this clear()
+		this _list free()
+		this _mutex free()
+		super()
+	}
+	add: func (object: T) {
+		this _mutex lock()
+		this _list add(object)
+		if (this _list count > this _maxCount)
+			this _free(this _list remove(0))
+		this _mutex unlock()
+	}
+	search: func (matches: Func (T) -> Bool) -> T {
```

Comment:
I guess we'll find out when that happens, I don't think it will be very common for us to recycle covers anyway.